compiler+runtime/src/cpp/jank/error/analyze.cpp

```diff
@@ -2,6 +2,12 @@
 
 namespace jank::error
 {
+  error_ptr
+  analysis_invalid_case(native_persistent_string const &message, read::source const &source)
+  {
+    return make_error(kind::analysis_invalid_def, message, source);
```

Comment:
Let's add a note suggesting that the dev use the `case` macro. They should only run into these errors if they use `case*` directly.

```cpp
return make_error(kind::analysis_invalid_def, message, source, note{ "Consider using the 'case' macro instead of using 'case*' directly", source });
```